### DISCUSSION TITLE
appPackage info

### DISCUSSION THREAD

**Discussion by FCMHUB** [Budibase/budibase#8266] [Help] [answered]
First of all, congratulations on this great tool.

I have a security-related question.
Is there any reason why the appPackage file has some back-end-related information sent to the browser? 
For instance, looking at the appPackage file we can see the datasources with their related queries exposed to the front end.

That brings me some thoughts.

- If this info is somehow used by the client, couldn't a man-in-the-middle attack change that file and grant some undesired database access?

Or

- If the info is not used by the client, shouldn't it be kept only on the back end?

Thanks in advance,

**Comment by melohagan** [comment, marked as answer]
Hey @FCMHUB 

Good observation! I've created an issue so that the developers can see it. https://github.com/Budibase/budibase/issues/8381

**Reply by shogunpurple** [reply]
[Should now be resolved](https://github.com/Budibase/budibase/issues/8523)
